aci/resource_aci_dhcpoptionpol.go

```diff
@@ -296,50 +293,55 @@ func resourceAciDHCPOptionPolicyUpdate(ctx context.Context, d *schema.ResourceDa
 	}
 
 	if d.HasChange("dhcp_option") {
-		dhcpOption := d.Get("dhcp_option_ids").([]interface{})
-		for _, val := range dhcpOption {
-			dhcpOptionDN := val.(string)
-			err := aciClient.DeleteByDn(dhcpOptionDN, "dhcpOption")
-			if err != nil {
-				return diag.FromErr(err)
+		old_options, new_options := d.GetChange("dhcp_option")
+		getDifference := differenceInMaps(old_options.(*schema.Set), new_options.(*schema.Set))
+		listDNToDelete := make([]string, 0)
+		for i := 0; i < len(getDifference); i++ {
```

Comment:
Why not use for ... := range ...  and use this for loop syntax?

Reply:
Changed the format!